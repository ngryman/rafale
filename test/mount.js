import test from 'ava'
import { nodeToJSON } from './helpers/to-json'
import { html, mount } from '../'

function macro(t, vtree) {
  mount(vtree, t.context)
  t.snapshot(nodeToJSON(t.context))
}

test.beforeEach(t => {
  const node = document.createElement('div')
  document.body.appendChild(node)
  t.context = node
})

test.afterEach(t => {
  t.context.remove()
})

test('mount a simple component', macro,
  html`<p>foo</p>`
)

test('mount a component with a primitive child', macro,
  html`<b>${'foo'}</b>`
)

test('mount a component with a nested component child', macro,
  html`<p>${
    html`<b>foo</b>`
  }</p>`
)

test('mount a component with a sequence child', macro,
  html`<p>${
    ['foo', 'bar'].map(text =>
      html`<b>${text}</b>`
    )
  }</p>`
)

test('mount a component with a primitive attribute', macro,
  html`<p id="${'foo'}"></p>`
)

test('mount a component with a primitive attribute without quotes', macro,
  html`<p id=${'foo bar'}></p>`
)

test('mount a component with mixed attributes', macro,
  html`<p id="${'foo'}" class="bar"></p>`
)

test('mount a component with a function attribute', macro,
  html`<button onclick=${_ => {}}></button>`
)

test('mount a component with dummy text nodes', macro,
  html`<b>
    ${'foo'}
  </b>`
)

test('mount on document.body by default', t => {
  mount(html`<b>foo</b>`)
  t.is(document.body.lastElementChild.textContent, 'foo')
  document.body.textContent = ''
})
