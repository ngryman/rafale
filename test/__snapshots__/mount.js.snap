// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mount a component with a function attribute 1`] = `
Object {
  "attributes": Array [],
  "children": Array [
    Object {
      "attributes": Array [],
      "children": Array [],
      "nodeName": "BUTTON",
      "onclick": undefined,
      "textContent": "",
    },
  ],
  "nodeName": "DIV",
  "onclick": undefined,
  "textContent": "",
}
`;

exports[`mount a component with a nested component child 1`] = `
Object {
  "attributes": Array [],
  "children": Array [
    Object {
      "attributes": Array [],
      "children": Array [],
      "nodeName": "P",
      "onclick": undefined,
      "textContent": "foo",
    },
  ],
  "nodeName": "DIV",
  "onclick": undefined,
  "textContent": "foo",
}
`;

exports[`mount a component with a primitive attribute 1`] = `
Object {
  "attributes": Array [],
  "children": Array [
    Object {
      "attributes": Array [
        Object {
          "name": "id",
          "value": "foo",
        },
      ],
      "children": Array [],
      "nodeName": "P",
      "onclick": undefined,
      "textContent": "",
    },
  ],
  "nodeName": "DIV",
  "onclick": undefined,
  "textContent": "",
}
`;

exports[`mount a component with a primitive attribute without quotes 1`] = `
Object {
  "attributes": Array [],
  "children": Array [
    Object {
      "attributes": Array [
        Object {
          "name": "id",
          "value": "foo bar",
        },
      ],
      "children": Array [],
      "nodeName": "P",
      "onclick": undefined,
      "textContent": "",
    },
  ],
  "nodeName": "DIV",
  "onclick": undefined,
  "textContent": "",
}
`;

exports[`mount a component with a primitive child 1`] = `
Object {
  "attributes": Array [],
  "children": Array [
    Object {
      "attributes": Array [],
      "children": Array [],
      "nodeName": "B",
      "onclick": undefined,
      "textContent": "foo",
    },
  ],
  "nodeName": "DIV",
  "onclick": undefined,
  "textContent": "foo",
}
`;

exports[`mount a component with a sequence child 1`] = `
Object {
  "attributes": Array [],
  "children": Array [
    Object {
      "attributes": Array [],
      "children": Array [
        Object {
          "attributes": Array [],
          "children": Array [],
          "nodeName": "B",
          "onclick": undefined,
          "textContent": "foo",
        },
        Object {
          "attributes": Array [],
          "children": Array [],
          "nodeName": "B",
          "onclick": undefined,
          "textContent": "bar",
        },
      ],
      "nodeName": "P",
      "onclick": undefined,
      "textContent": "foobar",
    },
  ],
  "nodeName": "DIV",
  "onclick": undefined,
  "textContent": "foobar",
}
`;

exports[`mount a component with dummy text nodes 1`] = `
Object {
  "attributes": Array [],
  "children": Array [
    Object {
      "attributes": Array [],
      "children": Array [],
      "nodeName": "B",
      "onclick": undefined,
      "textContent": "foo
  ",
    },
  ],
  "nodeName": "DIV",
  "onclick": undefined,
  "textContent": "foo
  ",
}
`;

exports[`mount a component with mixed attributes 1`] = `
Object {
  "attributes": Array [],
  "children": Array [
    Object {
      "attributes": Array [
        Object {
          "name": "id",
          "value": "foo",
        },
        Object {
          "name": "class",
          "value": "bar",
        },
      ],
      "children": Array [],
      "nodeName": "P",
      "onclick": undefined,
      "textContent": "",
    },
  ],
  "nodeName": "DIV",
  "onclick": undefined,
  "textContent": "",
}
`;

exports[`mount a simple component 1`] = `
Object {
  "attributes": Array [],
  "children": Array [
    Object {
      "attributes": Array [],
      "children": Array [],
      "nodeName": "P",
      "onclick": undefined,
      "textContent": "foo",
    },
  ],
  "nodeName": "DIV",
  "onclick": undefined,
  "textContent": "foo",
}
`;
